client/src/lib.js

```diff
@@ -112,6 +114,39 @@ class NFTStorage {
     }
   }
 
+  /**
+   * @param {API.Service} service
+   * @param {any} metadata
+   * @returns {Promise<API.CIDString>}
+   */
+  static async storeMetadata(service, metadata) {
+    /**
+     * @param {any} obj
+     * @returns {Promise<any>}
+     */
+    async function transform (obj) {
+      if (Array.isArray(obj)) {
+        return Promise.all(obj.map(transform))
+      }
+      if (obj instanceof File) {
+        return NFTStorage.storeDirectory(service, [obj])
+      }
+      if (obj instanceof Blob) {
+        return NFTStorage.storeBlob(service, obj)
+      }
+      if (typeof obj === 'object') {
+        const ents = await Promise.all(Object.entries(obj).map(async ([k, v]) => ([k, await transform(v)])))
+        return Object.fromEntries(ents)
+      }
+      return obj
+    }
+    metadata = await transform(metadata)
+    const cid = await NFTStorage.storeBlob(service, new Blob([cbor.encode(metadata)]))
+    console.log(cid)
+    const mh = CID.parse(cid).multihash
+    return CID.createV1(cbor.code, mh).toString()
```

Comment:
This is cool! I was thinking of doing something slightly different that is:

1. Create a content bundle like a car file so that
2. It would inline all the files
3. A JSON serialization written as `metadata.json`
4. A dag-cbor with pointers to all of the above (kind of like you do here)

Then we could turn return value into:

```js
{
   cid: CIDString // CID to the cbor with links to everything
   jsonURL: URL // ipfs://${cid} URL to a metadata.json
   data: JSON // JSON that written as `metadata.json`
}
```

That way:

1. Single NFT will correspond to single dag-cbor with links to all the relevant files and encoding al data.
2. We don't force IPLD, JSON representation of it is still there.
3. We do also proactively make things IPLD native, and lay the upgrade path to folks that want to.
4. Since we have all the data locally anyway, we include that in `data` so that URLs for all the files can be pulled out without another roundtrip.